compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2266,110 +2300,152 @@ static void setLoweringConfigForComputeOps(func::FuncOp entryPointFn,
   }
 
   auto ctx = entryPointFn.getContext();
-  TilingConfig tilingConfig(getLoweringConfig(rootOperation));
-  SmallVector<int64_t> distTileSizes, tileAndFuseSizes;
+  auto rootLoweringConfig = getLoweringConfig(rootOperation);
+  TilingConfig tilingConfig(rootLoweringConfig);
+  SmallVector<int64_t> distTileSizes, parallelVecTileSizes;
   if (tilingConfig.getNumTilingLevels() > 0) {
     distTileSizes = tilingConfig.getDistributionTileSizes();
   }
   if (tilingConfig.getNumTilingLevels() > 1) {
     // TODO: Handle scalable tiles.
-    std::tie(tileAndFuseSizes, std::ignore) =
+    std::tie(parallelVecTileSizes, std::ignore) =
         tilingConfig.getVectorCommonParallelSizes();
   }
 
-  // Multi-lowering config works only if all the operations can share the same
-  // distribution and TileAndFuse tile sizes.
+  size_t maxLoopNums = 0;
   for (auto op : computeOps) {
+    // Multi-lowering config works only if all the operations can share the same
+    // distribution and parallel tile sizes from the root op.
     auto iterTypes = cast<TilingInterface>(op).getLoopIteratorTypes();
     for (auto [idx, iterType] : llvm::enumerate(iterTypes)) {
-      if (idx >= tileAndFuseSizes.size())
+      if (idx >= parallelVecTileSizes.size())
         break;
       if (iterType == utils::IteratorType::parallel)
         continue;
-      if (distTileSizes[idx] || tileAndFuseSizes[idx])
+      if (distTileSizes[idx] || parallelVecTileSizes[idx])
         return;
     }
-  }
-
-  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
-  auto targetMLTransInfo =
-      TargetMLTransformInfo::getTargetMLTransformInfo(targetAttr);
+    maxLoopNums = std::max(maxLoopNums, iterTypes.size());
+  }
+
+  // Adjust the distribution tile sizes and join parallel vector tile sizes from
+  // other ops. The results of parallel vector tile sizes might overlap
+  // reduction dimensions on some ops, so it will be splitted into common vector
+  // tile sizes and inner vector tile sizes later.
+  //
+  // This step is to ensure all ops are using an equivalent set of parallel tile
+  // sizes.
+  //
+  // Here we use the assumption in FormDispatchRegions that all ops in a
+  // dispatch have identity mapping between their parallel dimensions. So we
+  // don't need to handle the permutation on dimensions between ops except for
+  // the pack op.
```

Comment:
Same, I think an example for this doc would be super helpful!

Reply:
Done.